automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/monitor/SubmissionDetailsPage.scala

```diff
@@ -77,8 +76,34 @@ class SubmissionDetailsPage(namespace: String, name: String, var submissionId: S
     }
   }
 
-  def abortSubmission(): Unit = {
+  def clickAbortButton(): AbortSubmissionConfirmationModal = {
     submissionAbortButton.doClick()
-    submissionAbortModalConfirmButton.doClick()
+    await ready new AbortSubmissionConfirmationModal()
+  }
+
+  def abortSubmission(): Unit = {
+    val confirmModal = clickAbortButton()
+    confirmModal.submit()
+  }
+
+  /**
+    * page class for Submission Abort Confirmation modal
+    */
+  class AbortSubmissionConfirmationModal(implicit webDriver: WebDriver) extends ConfirmModal {
```

Comment:
We shouldn't need this class. That's the main reason for having the confirm-modal distinction in the UI.

Reply:
hmm, it is to verify expected text displayed in expected Confirm modal. Displayed texts on different Confirm modals are going to be different.

Reply:
I feel like we don't need to test that. We should be testing broad behavior, not wording choice. At a high level, we click the "abort" button, and we click "confirm" on the modal that pops up.

Reply:
Hi Matt, I tested this to be sure it works. What are your thoughts on this? Thank you for your attention.

Reply:
Again, I don't want to subclass ConfirmModal. It's not designed to be subclassed.

Reply:
In that case, can you go ahead merge your PR https://github.com/broadinstitute/firecloud-ui/pull/1239? You have a good solution.